include/boost/gil/extension/numeric/convolve.hpp

```diff
@@ -373,7 +373,7 @@ void convolve_2d_impl(SrcView const& src_view, DstView const& dst_view, Kernel c
                         col_boundary >= 0 && col_boundary < src_view.width())
                     {
                         aux_total +=
-                            src_view(col_boundary, row_boundary) *
+                            src_view(col_boundary, row_boundary)[0] *
```

Comment:
Convolution algorithms only work on grayscale images? I am wondering what this extra [0] means... get the first channel?

Reply:
> I am wondering what this extra [0] means... get the first channel?

We are trying to access the pixel value of a specific channel from a channel specific view. [This](https://github.com/boostorg/gil/blob/c9b339e5d0b6a1d121ff9b37ae2d7544c534dc68/include/boost/gil/extension/numeric/convolve.hpp#L435) and [this](https://github.com/boostorg/gil/blob/c9b339e5d0b6a1d121ff9b37ae2d7544c534dc68/include/boost/gil/rasterization/ellipse.hpp#L137) may provide more context.